vmm/src/lib.rs

```diff
@@ -212,6 +212,46 @@ pub enum VmmActionError {
     VsockConfig(ErrorKind, VsockError),
 }
 
+// It's convenient to turn StartMicrovmErrors into VmmActionErrors directly.
+impl std::convert::From<StartMicrovmError> for VmmActionError {
+    fn from(e: StartMicrovmError) -> Self {
+        let kind = match e {
+            StartMicrovmError::ConfigureSystem(_) => ErrorKind::Internal,
```

Comment:
I think it would be more readable if you grouped together all the error types that match to `Internal` and `User` respectively, like:

```rust
match e {
    StartMicrovmError::ConfigureSystem(_)
    | StartMicrovmError::ConfigureVm(_)
    | StartMicrovmError::CreateRateLimiter(_) 
    | ...
    => ErrorKind::Internal,
...
```

Reply:
Yes, I thought about that too, and am currently leaning that way. The original grouping was to keep alphabetical order, but I now agree that grouping by kind makes more sense.